### ISSUE TITLE
Installing RiderLink to engine bricks UE5 on Mac

### ISSUE THREAD

**Issue by shadowmint** [JetBrains/UnrealLink#179]
After selecting 'Install RiderLink to engine' as per this forum thread:
https://forums.unrealengine.com/t/trouble-creating-c-project-in-ue5-ea/232968

It becomes impossible to build any project. Any attempt to build the project will result in:

> ERROR: Expecting to find a type to be declared in a module rules named ‘RD’ in UE5Rules

This effectively bricks your installation of UE5. Verify -> no problem. However, attempting any C++ build will result in this error.

### Fix

    cd /Users/Shared/Epic Games/UE_5.0/Engine/Plugins/Developer
    rm -rf RiderLink
    rm -rf RiderSourceCodeAccess

### Replication

- Load a c++ project in rider by selecting the uplugin
- Wait for the 'RiderLink plugin is required' popup
- Select 'Install plugin in Engine'
- Close rider
- Open the project in rider again
- Project cannot be built:

![image](https://user-images.githubusercontent.com/355760/166132968-3636b6d1-cfba-4228-8135-6756fc869833.png)

For more details:

    /Users/Shared/Epic\ Games/UE_5.0/Engine/Build/BatchFiles/Mac/GenerateProjectFiles.sh -project `pwd`/Jam.uproject -game
    open ./Jam.xcworkspace

Now attempt to run the project:

```
Setting up bundled DotNet SDK
Processing build for Target=JamEditor Platform=Mac Configuration=Development -architecture=x86_64 /Users/doug/dev/randomjam-2022-04/Jam/Jam.uproject 
Log file: /Users/doug/Library/Application Support/Epic/UnrealBuildTool/Log.txt
Creating makefile for ShaderCompileWorker (no existing makefile)
ERROR: Couldn't find target rules file for target 'ShaderCompileWorker' in rules assembly 'UE5Rules, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
       Location: /Users/Shared/Epic Games/UE_5.0/Engine/Intermediate/Build/BuildRules/UE5Rules.dll
       Target rules found:
Running dotnet Engine/Binaries/DotNET/UnrealBuildTool/UnrealBuildTool.dll JamEditor Mac Development /Users/doug/dev/randomjam-2022-04/Jam/Jam.uproject -architecture=x86_64
Log file: /Users/doug/Library/Application Support/Epic/UnrealBuildTool/Log.txt
Using 'git status' to determine working set for adaptive non-unity build (/Users/doug/dev/randomjam-2022-04).
Invalidating makefile for JamEditor (RiderSourceCodeAccess.Build.cs deleted)
ERROR: Expecting to find a type to be declared in a module rules named 'RD' in UE5Rules, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null.  This type must derive from the 'ModuleRules' type defined by Unreal Build Tool.
Command /Users/Shared/Epic Games/UE_5.0/Engine/Build/BatchFiles/Mac/XcodeBuild.sh failed with exit code 6
```

UE5 is now bricked. It cannot compile any project until the plugin is deleted from the engine folder.

### Environment

> macOS Monterey 12.3.1, Max Studio (2022)

> unreal 5.0.1 

> rider 2022.1.1

**Comment by DecoyRS** [comment]
Thank you for the detailed description, but this issue is on the Unreal Engine side and I'm sorry that you've spent so much time describing it, because the description really looks great and formatting is top notch... :(
Last builds of Rider (not Rider for Unreal Engine) disable option to install plugin to the pre-built Engine. You will still be able to install it to Engine. 

What's actually happening?
Starting with UE5, Engine started to pack all the Engine plugins into UE5Rules.dll and breaks if it can't find a plugin in this dll. So far I couldn't find a way to generate a new UE5Rules file and the only workaround is to place plugins into Plugins/Marketplace (because there's an exception for this folder to allow thirdparty plugins from marketplace) , but I believe this is not the right place for the RiderLink (though I have plans to have a customizable install location, but priority is really low right now).

**Comment by shadowmint** [comment]
Fair enough. If this is an issue on the Unreal side, I'm happy for this to be closed.

> Last builds of Rider (not Rider for Unreal Engine) disable option to install plugin to the pre-built Engine. 

However, I'd point out I'm not using Rider for Unreal Engine, I'm using the full Rider 2022.1.1 with the plugin from the marketplace:

![image](https://user-images.githubusercontent.com/355760/166144164-c69b96c0-6165-4e8a-af77-bc9a5df8400e.png)

...and I assure you, it automatically popups up and asks me to install the plugin into the engine. 

I definitely feel like (regardless of the issue in Unreal), the plugin should absolutely not suggest any 'one-click' options that permanently render your install of UE5 broken. ...but, if that's an upstream change that simply hasn't been rolled out yet, by all means close this issue.
